### DISCUSSION TITLE
Is it feasible to code a version of ASM in Python?

### DISCUSSION THREAD

**Discussion by emullallphotography** [community/community#70233] [Programming Help]
### Body

Hi,

I am a Python programmer who tried ASM and wanted to build a simplified version of it in Python. Is that even feasible?

### Guidelines

- [x] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by emullallphotography** [comment]
I already thought of basic functions: add, move, eax, _entry, val, edx, esi, edi, edx, etc

_entry creates an entry point
Val is used for moving values into registers: move val[5] into eax
into is used to specify which register the value moves into
Add is used to add two numbers: add 1, 2 into eax. This function will make the CPU take to values(1 and 2), add them than store the sum into the specified register(eax)
move is used to move data around

**Comment by davea42** [comment]
Please consider the effect of changing a register via your code (that you plan to write).  Any register change could crash  the python interpreter (or worse), assuming  your intent is to change the register live.  Perhaps I just don't understand your intent.

**Reply by emullallphotography** [reply]
What is the worst thing to happen?

**Reply by davea42** [reply]
I suppose the most accurate answer would be: 'I have no idea what could happen."  On the other hand, note that programs
to do arbitrary things in a running program (given sufficient permission is available) already exist.  
For example the GNU debugger gdb.      So given the person running gdb has  permission to intercept a program
it can do what you refer to.  Which has been true for decades without bad result beyond interfering with the program being
controlled ...AFAIK.  You might find it enlightening to research how debuggers work and look into how they are coded.
